src/rpc/misc.cpp

```diff
@@ -570,6 +600,7 @@ static const CRPCCommand commands[] =
 
     /* Not shown in help */
     { "hidden",             "setmocktime",            &setmocktime,            {"timestamp"}},
+    { "hidden",             "mockscheduler",          &mockscheduler,          {"delta_time"}},
```

Comment:
Follow up, would be cool to skip this when !regtest and in `mockscheduler` would be `assert(Params().MineBlocksOnDemand())`.

Reply:
I think it's reasonable for @amitiuttarwar to just copy exactly what setmocktime is doing in this regard.

Reply:
Huh? `static const commands[]=` will get done prior to working out what chain is in use, so you'd have to do a much more serious rework to remove it from the list of rpcs entirely.

Reply:
It's a hidden RPC so I don't think this matters much.

Reply:
@ajtowns I was thinking that `CRPCTable::execute` could take into account some `CRPCCommand` flag and then it would result in `RPC_METHOD_NOT_FOUND`.